announcement/models/res_users.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2022 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class ResUsers(models.Model):
+    _inherit = "res.users"
+
+    pendant_announcement_ids = fields.Many2many(
+        comodel_name="announcement", relation="pendant_announcement_user_rel",
+    )
+    read_announcement_ids = fields.Many2many(
+        comodel_name="announcement", relation="read_announcement_user_rel",
+    )
+    popup_announcements = fields.Boolean()
+
+    @api.model
+    def announcement_user_count(self):
+        """The js widget gathers the announcements from this method"""
+        now = fields.Datetime.now()
+        group_announcements = self.env["announcement"].search_read(
+            [
+                ("announcement_type", "=", "user_group"),
+                "|",
+                ("notification_date", "<=", now),
+                ("notification_date", "=", False),
+                "|",
+                ("notification_expiry_date", ">=", now),
+                ("notification_expiry_date", "=", False),
+                ("id", "not in", self.env.user.read_announcement_ids.ids),
+            ],
+            ["user_group_ids"],
+        )
+        announcements = self.env["announcement"].browse(
+            {
+                x["id"]
+                for x in group_announcements
+                if any(
+                    [g for g in x["user_group_ids"] if g in self.env.user.groups_id.ids]
+                )
+            }
+        )
+        announcements |= self.env.user.pendant_announcement_ids.filtered(
+            lambda x: (not x.notification_date or x.notification_date <= now)
+            and (not x.notification_expiry_date or x.notification_expiry_date >= now)
+        ).sorted(lambda k: k.sequence)
+        return announcements.read()
```

Comment:
here you are reading again everything from the db, can't you read only what is needed from the recordset?